sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowJson.java

```diff
@@ -148,12 +152,22 @@ private static ImmutableList<UnsupportedField> findUnsupportedFields(
     TypeName fieldTypeName = fieldType.getTypeName();
 
     if (fieldTypeName.isCompositeType()) {
-      return fieldType.getRowSchema().getFields().stream()
-          .flatMap(
-              (field) ->
-                  findUnsupportedFields(field.getType(), fieldName + "." + field.getName())
-                      .stream())
-          .collect(toImmutableList());
+      if (fieldTypeName == TypeName.ROW) {
+        return fieldType.getRowSchema().getFields().stream()
+            .flatMap(
+                (field) ->
+                    findUnsupportedFields(field.getType(), fieldName + "." + field.getName())
+                        .stream())
+            .collect(toImmutableList());
+      } else if (fieldTypeName == TypeName.MAP) {
```

Comment:
I am not sure if this could change current behavior. To be safe and avoid breaking change how about

```
if (fieldTypeName == TypeName.MAP) {
  ....
  return ...
}
<current code>
```